trin-history/src/jsonrpc.rs

```diff
@@ -35,236 +37,262 @@ impl HistoryRequestHandler {
+                    let enr = convert_enr(enr);
+
+                    let response = match self
+                        .network
+                        .overlay
+                        .send_find_content(enr, content_key.into())
+                        .await
+                    {
+                        Ok(content) => match content.try_into() {
+                            Ok(val) => Ok(val),
+                            Err(_) => Err("Content response decoding error".to_string()),
                         },
-                        Err(msg) => Err(format!("Invalid FindNodes params: {:?}", msg)),
+                        Err(msg) => Err(format!("FindContent request timeout: {:?}", msg)),
                     };
                     let _ = request.resp.send(response);
                 }
-                HistoryEndpoint::Offer => {
-                    let response = match OfferParams::<HistoryContentKey>::try_from(request.params)
+                HistoryEndpoint::FindNodes(enr, distances) => {
+                    let enr = convert_enr(enr);
+
+                    let response = match self.network.overlay.send_find_nodes(enr, distances).await
                     {
-                        Ok(params) => {
-                            let content_key = params.content_key.clone();
-                            let content_items = vec![(content_key, params.content)];
-                            let num_peers = self.network.overlay.propagate_gossip(content_items);
-                            Ok(num_peers.into())
-                        }
-                        Err(msg) => Err(format!("Invalid Offer params: {:?}", msg)),
+                        Ok(nodes) => Ok(json!({
+                            "total": nodes.total,
+                            "enrs":  nodes.enrs.into_iter().map(|enr| EthEnr::from_str(&enr.to_base64()).unwrap()).collect::<Vec<EthEnr>>()
+                        })),
+                        Err(msg) => Err(format!("FindNodes request timeout: {:?}", msg)),
                     };
                     let _ = request.resp.send(response);
                 }
-                HistoryEndpoint::SendOffer => {
-                    let response = match SendOfferParams::try_from(request.params) {
-                        Ok(val) => {
-                            let content_keys =
-                                val.content_keys.iter().map(|key| key.to_vec()).collect();
+                HistoryEndpoint::Offer(content_key, content_item) => {
+                    match convert_content_key(&content_key) {
+                        Ok(content_key) => {
+                            let response = {
+                                let content_items = vec![(content_key, content_item.into())];
+                                let num_peers =
+                                    self.network.overlay.propagate_gossip(content_items);
+                                Ok(num_peers.into())
+                            };
```

Comment:
the following is more clear to me:

```suggestion
                                let content_items = vec![(content_key, content_item.into())];
                                let num_peers =
                                    self.network.overlay.propagate_gossip(content_items);
                                let response = Ok(num_peers.into());
```